cdap-app-fabric/src/main/java/co/cask/cdap/route/store/ZKRouteStore.java

```diff
@@ -120,11 +121,9 @@ public RouteConfig fetch(final ProgramId serviceId) {
   private RouteConfig getConfig(ProgramId serviceId, Future<RouteConfig> future) {
```

Comment:
Can't comment in the `fetch` method above (line 109), hence comment in here. For the fetch method to perform faster, it's better to first do a `get` on the concurrent map. Only if it returns `null`, we'll do the `putIfAbsent` logic as you have now. Otherwise, we are creating a new `SettableFuture` object on every call.